### PR TITLE
Don't attach the control plane security group to managed nodegroups

### PR DESCRIPTION
**Pull request by tompiscitell**
### Description

When creating Managed Nodegroups from an imported cluster, the control plane security group was getting attached to the worker nodes. This security group is supposed to be attached to the control plane ENIs according to [the EKS docs](https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html#control-plane-worker-node-sgs). With this PR, only the cluster security group will get attached to Managed Nodegroups. This is consistent with the behavior of adding Managed Nodegroups to clusters created by eksctl.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [x] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by Himangini** [comment]
@tompiscitell thanks for the nicely written PR. Core team will review it soon 👍🏻

**Comment by Skarlso** [comment]
Hi @tompiscitell.

That document says earlier than 1.14. What makes you think this is not required for versions above 1.14?

File: pkg/vpc/importer.go [hunk 1]
```diff
@@ -119,19 +119,18 @@ func (si *SpecConfigImporter) SharedNodeSecurityGroup() *gfnt.Value {
 	return si.ClusterSecurityGroup()
 }
 
-// SecurityGroups returns a gftn slice of the ControlPlaneSecurityGroup
-// and the ClusterSecurityGroup
+// SecurityGroups returns a gfnt slice of the ClusterSecurityGroup
 func (si *SpecConfigImporter) SecurityGroups() gfnt.Slice {
-	return gfnt.Slice{si.ControlPlaneSecurityGroup(), si.ClusterSecurityGroup()}
+	return gfnt.Slice{si.ClusterSecurityGroup()}
```
Comment (Skarlso) [inline pkg/vpc/importer.go:124]: This change only affects unowned clusters. For Owned clusters we are already doing this. I except this is actually not doing anything, but I have to further check the code.

**Comment by Skarlso** [comment]
@tompiscitell Ok, now that I understand the code a bit better, let me summarise. This change is only applicable to unowned clusters. Meaning, it's a cluster that `eksctl` has not created. For managed clusters we already do the right thing. I'm okay with this change, but please paste in manual tests and steps and outcome so we can see that you tried this code change and an unowned cluster still works in creating managed nodegroups and communication is not disrupted. 

So what you need:
- create an EKS cluster but not by using `eksctl`
- import that cluster
- create a managed nodegroup
- show results

Thanks! :)

**Review by Skarlso — CHANGES_REQUESTED** [review]
Request Manual Test steps then LGTM.

**Review by Skarlso — APPROVED** [review]

File: pkg/vpc/importer.go (hunk 1 above)
Comment (aclevername) [reply pkg/vpc/importer.go:124]: > This change only affects unowned clusters. For Owned clusters we are already doing this. I except this is actually not doing anything, but I have to further check the code.

why would this change only affect unowned? I'm slightly confused :/

Comment (aclevername) [reply pkg/vpc/importer.go:124]: ahh makes sense, thank you!

**Review by aclevername — APPROVED** [review]

**Pull request merged**